src/test/scala/chiseltest/experimental/tests/AssertTest.scala

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiseltest.experimental.tests
+
+import chisel3._
+import chisel3.tester._
+import chiseltest.VerilatorBackendAnnotation
+import org.scalatest.freespec.AnyFreeSpec
+
+class AssertFail extends Module {
+  val io = IO(new Bundle {
+    val input = Input(UInt(2.W))
+  })
+
+  printf("io.input is %d\n", io.input) // prints 3 and then 0
+
+  assert(io.input === 3.U) // fails, apparently on the second cycle
+}
+
+class AssertTest extends AnyFreeSpec with ChiselScalatestTester {
```

Comment:
I think this test case needs a comment, because the behavior it's testing is non-obvious. Something along the lines of that because pokes go out of scope at the end of the main test body, there should be no trailing clock edge (and no asserts evaluated) after the test ends.

Reply:
Good point, better comment added